src/libraries/System.IO/tests/TextReader/TextReaderTests.cs

```diff
@@ -54,6 +55,23 @@ public async Task ReadToEndAsync()
             }
         }
 
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]
+        public async Task ReadToEndAsync_WithCancellationToken()
+        {
+            using var tr = new CharArrayTextReader(TestDataProvider.LargeData);
+            var result = await tr.ReadToEndAsync(default);
+            Assert.Equal(5000, result.Length);
+        }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]
+        public async Task ReadToEndAsync_WithCanceledCancellationToken()
+        {
+            using var tr = new CharArrayTextReader(TestDataProvider.LargeData);
+            using var cts = new CancellationTokenSource();
+            cts.Cancel();
+            await Assert.ThrowsAnyAsync<OperationCanceledException>(async () => await tr.ReadToEndAsync(cts.Token));
```

Comment:
we can also verify the value of the `TaskCanceledException.CancellationToken`:

```suggestion
            using var cts = new CancellationTokenSource();
            cts.Cancel();
            CancellationToken token = cts.Token;

            TaskCanceledException ex = await Assert.ThrowsAsync<OperationCanceledException>(async () => await tr.ReadToEndAsync(token));
            Assert.Equal(token, ex.CancellationToken);
```